File: python/table4.py
from getOptimalQuadPoints import getOptimalQuadPoints
import numpy as np
import sys

def get_knot(p,k,nel):
    ans = [0] * (k+1)
    for i in range(nel+1):
        ans += [i] * (p-k)
    ans += [nel] * (k+1)
    return ans

nel = 128
for p in range(8,17):
    for k in range(p):
        knot = get_knot(p,k,nel)
        w,x,rec,it = getOptimalQuadPoints(knot,p)
        print(f'{rec:3d}  ', end='')
    print('')

File: matlab/table4.m

nel = 128;
for p=8:16
    for k=0:p-1
        knot = [];
        for m=1:p-k % create multiple knots for reduced continuity
            knot = [knot, 0:nel];
        end
        for m=p-k:p % make sure start and end knot are repeated p+1 times
            knot = [knot, 0, nel];
        end
        knot = sort(knot);
        [w, x, rec, it] = getOptimalQuadPoints(knot, p);
        fprintf('%3d  ', rec);
    end
    fprintf('\n');
end
